src/lib/transition_frontier_controller_tests/stubs.ml

```diff
@@ -0,0 +1,346 @@
+        ~self:(Public_key.compress largest_account_public_key)
+        ~transactions_by_fee:transactions ~get_completed_work
+    in
+    let%bind ( `Hash_after_applying next_staged_ledger_hash
+             , `Ledger_proof ledger_proof_opt
+             , `Staged_ledger _ ) =
+      Staged_ledger.apply_diff_unchecked parent_staged_ledger
+        staged_ledger_diff
+      |> Deferred.Or_error.ok_exn
+    in
+    let previous_transition_with_hash =
+      Transition_frontier.Breadcrumb.transition_with_hash parent_breadcrumb
+    in
+    let previous_protocol_state =
+      With_hash.data previous_transition_with_hash
+      |> External_transition.Verified.protocol_state
+    in
+    let previous_ledger_hash =
+      previous_protocol_state |> Protocol_state.blockchain_state
+      |> Protocol_state.Blockchain_state.ledger_hash
+    in
+    let next_ledger_hash =
+      Option.value_map ledger_proof_opt
+        ~f:(fun proof ->
+          Ledger_proof.statement proof |> Ledger_proof.statement_target )
+        ~default:previous_ledger_hash
+    in
+    let next_blockchain_state =
+      Blockchain_state.create_value ~timestamp:(Block_time.now ())
+        ~ledger_hash:next_ledger_hash
+        ~staged_ledger_hash:next_staged_ledger_hash
+    in
+    let previous_state_hash =
+      Consensus.Mechanism.Protocol_state.hash previous_protocol_state
+    in
+    let consensus_state =
+      make_next_consensus_state ~snarked_ledger_hash:previous_ledger_hash
+        ~previous_protocol_state:
+          With_hash.{data= previous_protocol_state; hash= previous_state_hash}
+    in
+    let protocol_state =
+      Protocol_state.create_value ~previous_state_hash
+        ~blockchain_state:next_blockchain_state ~consensus_state
+    in
+    let next_external_transition =
+      External_transition.create ~protocol_state
+        ~protocol_state_proof:Proof.dummy
+        ~staged_ledger_diff:(Staged_ledger_diff.forget staged_ledger_diff)
+    in
+    (* We manually created a verified an external_transition *)
+    let (`I_swear_this_is_safe_see_my_comment
+          next_verified_external_transition) =
+      External_transition.to_verified next_external_transition
+    in
+    let next_verified_external_transition_with_hash =
+      With_hash.of_data next_verified_external_transition
+        ~hash_data:
+          (Fn.compose Consensus.Mechanism.Protocol_state.hash
+             External_transition.Verified.protocol_state)
+    in
+    match%map
+      Transition_frontier.Breadcrumb.build ~logger ~parent:parent_breadcrumb
+        ~transition_with_hash:next_verified_external_transition_with_hash
+    with
+    | Ok new_breadcrumb -> new_breadcrumb
+    | Error (`Fatal_error exn) -> raise exn
+    | Error (`Validation_error e) ->
+        failwithf !"Validation Error : %{sexp:Error.t}" e ()
+
+let create_root_frontier ~logger : Transition_frontier.t Deferred.t =
+  let accounts = Genesis_ledger.accounts in
```

Comment:
maybe later we can generate genesis ledgers as well instead of always starting from the same state.